tools/pybind11Common.cmake

```diff
@@ -174,11 +174,14 @@ endif()
 if(PYBIND11_NOPYTHON)
   set(_pybind11_nopython ON)
 elseif(
-  _pybind11_missing_old_python STREQUAL "NEW"
-  OR PYBIND11_FINDPYTHON
-  OR Python_FOUND
-  OR Python2_FOUND
-  OR Python3_FOUND)
+  NOT (DEFINED PYBIND11_FINDPYTHON AND NOT PYBIND11_FINDPYTHON)
```

Comment:
I find the long combination of conditions difficult to parse/understand.

What do you think about:

1. Tiny rewrite of the new additions:

```
(NOT DEFINED PYBIND11_FINDPYTHON OR PYBIND11_FINDPYTHON)
```

2. Adding a comment, similar to what you wrote in the PR description?